feat: DIA-1926: return annots/preds in sample data

pakelley opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/13814502645)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/427
<!-- Follow Merge -->

-------

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Adds sample annotations/predictions to sample task, when it can.

pakelley committed:
feat: return annots/preds in sample data

pakelley committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13711519961

matt-bernstein reviewed (commented):
Have you checked that this doesn't cause regressions or perf issues for when this endpoint is used to populate the label interface preview? I was imagining to have some kind of use_v2: bool = False param in the endpoint that would isolate it to only the new widget by default. If it just works for everything, then amazing, just gotta call out the extr test surface for QA

pakelley committed:
lint

pakelley committed:
lint

pakelley committed:
make annots/preds optional

pakelley committed:
lint

pakelley committed:
lint

pakelley committed:
lint

pakelley committed:
Merge branch 'develop' into 'fb-dia-1926'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13814502645

pakelley committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13814502645
